ms2query/ms2library.py

```diff
@@ -235,23 +236,27 @@ def _get_analog_search_scores(self,
             spectrum_id = query_spectrum.get(
                 self.settings["spectrum_id_column_name"])
             query_spectrum_parent_mass = query_spectrum.get("parent_mass")
+            results_table = ResultsTable(preselection_cut_off=preselection_cut_off,
```

Comment:
Might be an option to pass on a selection of the ms2ds_scores already (the ones related to this query spectrum)